### PR TITLE
Add doc strings for Update methods and Feedback methods / types

### PR DESCRIPTION
**Pull request by christothes**

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Metrics/MetricPeriodFeedback.cs [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+        /// <param name="metricId"> The metric unique id. </param>
+        /// <param name="dimensionFilter"> The <see cref="FeedbackDimensionFilter"/> to apply to the feedback. </param>
+        /// <param name="value"> The <see cref="PeriodFeedbackValue"/> for the feedback. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="metricId"/>, <paramref name="dimensionFilter"/>, or <paramref name="value"/> is null. </exception>
+        public MetricPeriodFeedback(string metricId, FeedbackDimensionFilter dimensionFilter, PeriodFeedbackValue value) : base(metricId, dimensionFilter)
+        {
+            if (metricId == null)
+            {
+                throw new ArgumentNullException(nameof(metricId));
+            }
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Metrics/MetricPeriodFeedback.cs:25]: We should have an empty-check for `metricId` (and update documentation accordingly).

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 2]
```diff
@@ -475,12 +475,16 @@ Page<EnrichmentStatus> NextPageFunc(string nextLink, int? pageSizeHint)
...
         /// </summary>
-        /// <param name="metricId"></param>
-        /// <param name="options"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <param name="metricId">The Id of the metric.</param>
+        /// <param name="options">The optional <see cref="GetMetricFeedbacksOptions"/> containing the options to apply to the request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>
+        /// A <see cref="AsyncPageable{T}"/> containing the collection of <see cref="MetricFeedback"/>s.
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:484]: `A` => `An` ?

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 3]
```diff
@@ -653,15 +668,17 @@ public virtual Response<MetricFeedback> CreateMetricFeedback(MetricFeedback feed
...
-        /// <param name="feedbackId"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <param name="feedbackId">The Id of the <see cref="MetricFeedback"/>.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>
+        /// A <see cref="Response{T}"/> containing the created <see cref="MetricFeedback"/>s.
+        /// </returns>
+        /// <exception cref="ArgumentException"><paramref name="feedbackId"/> is empty or not a valid GUID.</exception>
+
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:679]: Missing `ArgumentNullException` scenario.
nit: extra blank line.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs [hunk 4]
```diff
@@ -679,11 +696,14 @@ public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackAsync(strin
...
-        /// <param name="feedbackId"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <param name="feedbackId">The Id of the <see cref="MetricFeedback"/>.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>
+        /// A <see cref="Response{T}"/> containing the created <see cref="MetricFeedback"/>s.
+        /// </returns>
+        /// <exception cref="ArgumentException"><paramref name="feedbackId"/> is empty or not a valid GUID.</exception>
         public virtual Response<MetricFeedback> GetMetricFeedback(string feedbackId, CancellationToken cancellationToken = default)
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs:707]: Missing `ArgumentNullException` scenario here as well.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 5]
```diff
@@ -351,12 +351,16 @@ public virtual Response<DataFeed> CreateDataFeed(string dataFeedName, DataFeedSo
...
 
         /// <summary>
-        ///
+        /// Updates an existing <see cref="DataFeed"/>.
         /// </summary>
-        /// <param name="dataFeedId"></param>
-        /// <param name="dataFeed"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <param name="dataFeedId">The Id of the existing <see cref="AlertingHook"/> to update.</param>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:356]: It should be existing `DataFeed`.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs [hunk 6]
```diff
@@ -381,12 +385,16 @@ public virtual async Task<Response> UpdateDataFeedAsync(string dataFeedId, DataF
...
 
         /// <summary>
-        /// ///
+        /// Updates an existing <see cref="DataFeed"/>.
         /// </summary>
-        /// <param name="dataFeedId"></param>
-        /// <param name="dataFeed"></param>
-        /// <param name="cancellationToken"></param>
-        /// <returns></returns>
+        /// <param name="dataFeedId">The Id of the existing <see cref="AlertingHook"/> to update.</param>
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs:390]: `DataFeed` here as well.

**Review by kinelski — APPROVED** [review]
LGTM.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Metrics/MetricCommentFeedback.cs [hunk 7]
```diff
@@ -37,7 +37,17 @@ internal MetricCommentFeedback(string metricId, FeedbackDimensionFilter dimensio
         }
 
         /// <summary>
-        /// The comment.
+        /// The start timestamp of feedback time range.
+        /// </summary>
+        public DateTimeOffset? StartTime { get; set; }
+
+        /// <summary>
+        /// The end timestamp of feedback timerange. When this is equal to <see cref="StartTime"/> it indicates a single timestamp.
```
[the remark this answers is not in the document]
Comment (christothes) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Metrics/MetricCommentFeedback.cs:45]: I don't know - this is actually taken from the swagger and slightly edited.

**Pull request merged**